### ISSUE TITLE
Validations from "subform" disregarded by master form

### ISSUE THREAD

**Issue by msageryd** [final-form/react-final-form#673]
I'm not sure if this is a bug report, feature request or a support question. I'll post it here for starters.

### Subform

What I'm calling "subform" is the following.

A `<Form>` has a button for navigating to a "subform". The subform component gets a `form` from the master forms' render function, like this:
```
<Form
  render={({ handleSubmit, form }) => {
    <Button
      onPress={() => props.navigation.navigate('SubForm', { form })}
    >
      Open subform
    </Button>
...
```

The subform uses the handed form object like this:
```
function SubForm(props) {
  const form = props.navigation.getParam('form');
  return (
    <Form
      form={form}
      render={() => {
...
```

I'm using react-navigation in the example, but the point is that I'm marshalling the form to the subform, i.e. the subform operates on the same form object.

If I submit the master form, I also get the values from the subform as expected. 
But as soon as I go back to the master form, the form state changes to **valid** even if there are validation errors in the subform.

If I add a field with the same name in the master form it works as expected, even if I only edit the field from the subform.

### What is the expected behavior?

Submit should be prohibited when invalid fields exists in subforms (it's the same form object). I.e. FormSpy.props.valid should remain false if a subform is invalid.

### What's your environment?

final-form: 4.18.5
react-final-form: 6.3.0
react-native: 0.60

**Comment by msageryd** [comment]
Does anyone have any input on this? A workaround would probably be to create all fields in the master form, but make them invisible. But this would be kind of hacky.

Marshalling the form to subforms fits perfect for my use case. But I might have to high expectations on this functionality. What behaviour should I expect from the above example?

**Comment by msageryd** [comment]
I just realised that context is used to propagate data to `useField`, etc. I solved my problem by wrapping my form in a Portal. This way the context data is kept even after navigating to a subform (since I'm navigating with react-navigation, the new screen is not mounted in the same context as the screen with my master form).

I'm not sure if this is a good solution. I'll have to be careful to not mount another final-form while having my "master" form mounted in a Portal. It would be really messy.

I could solve this by using modals for the subform instead of navigating to them. Bu I really dislike modals for number of reasons. I think I'll give the portal concept a shot.

I'd really appreciate any thoughts on this.
